### PR TITLE
Update tests

### PR DESCRIPTION
**Pull request by roxblnfk**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌

Update psalm and phpunit; update tests and fix bug with deprecated doctrine/annotations function

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -21,28 +21,28 @@
...
     },
     "require-dev": {
         "cycle/entity-behavior": "^1.0",
         "jetbrains/phpstorm-attributes": "^1.0",
-        "phpunit/phpunit": "^9.5",
+        "phpunit/phpunit": "^9.6 || ^10",
         "roave/infection-static-analysis-plugin": "^1.25",
         "spatie/phpunit-watcher": "^1.23",
-        "vimeo/psalm": "^4.22|^5.4",
+        "vimeo/psalm": "^4.30 || ^5.7",
```
Comment (vjik) [inline composer.json:45]:
```suggestion
        "vimeo/psalm": "^4.30|^5.7",
```

https://github.com/yiisoft/docs/blob/master/000-packages.md#composerjson

File: composer.json [hunk 2]
```diff
@@ -21,28 +21,28 @@
...
-        "yiisoft/friendly-exception": "^1.0",
+        "yiisoft/friendly-exception": "^1.1",
         "yiisoft/injector": "^1.1",
         "yiisoft/yii-console": "^2.0"
     },
     "require-dev": {
         "cycle/entity-behavior": "^1.0",
         "jetbrains/phpstorm-attributes": "^1.0",
-        "phpunit/phpunit": "^9.5",
+        "phpunit/phpunit": "^9.6 || ^10",
```
Comment (vjik) [inline composer.json:42]:
```suggestion
        "phpunit/phpunit": "^9.6|^10",
```

https://github.com/yiisoft/docs/blob/master/000-packages.md#composerjson

File: psalm.xml [hunk 3]
```diff
@@ -4,6 +4,8 @@
     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
     xmlns="https://getpsalm.org/schema/config"
     xsi:schemaLocation="https://getpsalm.org/schema/config vendor/vimeo/psalm/config.xsd"
+    findUnusedCode="false"
+    findUnusedBaselineEntry="false"
```
Comment (vjik) [inline psalm.xml:8]:
```suggestion
    findUnusedBaselineEntry="true"
```

**Review by vjik — CHANGES_REQUESTED** [review]

File: psalm.xml (hunk 3 above)
Comment (roxblnfk) [reply psalm.xml:8]: There is no baseline

**Review by samdark — APPROVED** [review]

**Pull request merged**